xarray/core/groupby.py

```diff
@@ -259,6 +259,12 @@ class GroupBy:
         "_stacked_dim",
         "_unique_coord",
         "_dims",
+        "_dask_groupby_kwargs",
```

Comment:
Chose to add some extra private attributes instead of fully refactoring the constructor (which is a big change!). We'll have to refactor the constructor to enable grouping by multiple variables, so we could do that in a future PR.